editor/import/3d/scene_import_settings.cpp

```diff
@@ -1585,9 +1585,13 @@ void SceneImportSettingsDialog::_save_dir_confirm() {
 			continue; //ignore
 		}
 		String path = item->get_text(1);
+		String uid_path = path;
 		if (!path.is_resource_file()) {
```

Comment:
`uid://` will return false.

Reply:
Ah I had the if statements in the wrong order. I didn't notice because the first time resources are extracted, they are always res://
Also found a related bug with material uid assignment. I realized for materials we can simply use the standard flow of ResourceSaver::save() -> ResourceUID::path_to_uid(). No need to complicate things by generating a UID first and then calling set_uid.

The material logic in scene_import_settings should be much simpler now